pkg/fqdn/namemanager/gc_cache.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package namemanager
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// DefaultGCCacheSize is the default size of the GC cache
+const DefaultGCCacheSize = 10000
+
+// GCCache tracks recently garbage-collected FQDN entries.
+type GCCache struct {
+	cache *lru.Cache[string, time.Time]
+	mu    sync.RWMutex
+}
+
+// GCCacheEntry represents a garbage-collected FQDN entry.
+type GCCacheEntry struct {
+	FQDN                  string    `json:"fqdn"`
+	GarbageCollectionTime time.Time `json:"garbage-collection-time"`
+}
+
+// NewGCCache creates a new GCCache with the specified size.
+func NewGCCache(size int) (*GCCache, error) {
+	cache, err := lru.New[string, time.Time](size)
```

Comment:
Let's just panic here if `lru.New` returns an error (this only happens if size is less than zero, which is impossible here)